runner.go

```diff
@@ -0,0 +1,103 @@
+package async
+
+import (
+	"context"
+	"sync"
+)
+
+type Runner struct {
+	sync.Mutex
+	tasks      []Task
+	errs       []error
+	limit      int
+	waitErrors bool
+}
+
+// NewRunner creates a new task manager to control async functions
+func NewRunner(tasks ...Task) *Runner {
+	return &Runner{
+		tasks: tasks,
+		limit: len(tasks),
+	}
+}
+
+// WaitErrors if, if active, will wait for the error response from all functions
```

Comment:
```suggestion
// WaitErrors tells the runner to wait for the response from all functions instead of cancelling them all when the first error occurs.
```